core/store/src/trie/state_parts.rs

```diff
@@ -493,8 +493,13 @@ impl Trie {
         part_id: PartId,
         part: PartialState,
     ) -> ApplyStatePartResult {
-        Self::apply_state_part_impl(state_root, part_id, part)
-            .expect("apply_state_part is guaranteed to succeed when each part is valid")
+        Self::apply_state_part_impl(state_root, part_id, part).unwrap_or_else(|_| {
```

Comment:
nit: I recall some Results have a context method, if it works as I think it does it may be a bit nicer.